### PR TITLE
For #7586: Add email & call actions to context menu

### PR DESCRIPTION
**Pull request by sblatz**
---

**I know I need to add tests for this, but I wanted to make sure this approach was okay before doing so**

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by gabrielluong** [comment]
This looks pretty dam good.

**Comment by sblatz** [comment]
Awesome! Will fix merge conflicts & add tests 😄

**Comment by sblatz** [comment]
This should be ready to review now, @csadilek !

**Comment by sblatz** [comment]
It seems like the tests don't like `Patterns.EMAIL_ADDRESS.matcher(selectedText).matches()`

Any ideas how to get around this? When I remove the `&& Patterns` part of the conditional, the tests don't crash, but they give incorrect results (since that portion of the conditional is necessary for proper functioning).

@csadilek 

they give me this output

```

java.lang.NullPointerException
	at mozilla.components.feature.contextmenu.DefaultSelectionActionDelegate.isActionAvailable(DefaultSelectionActionDelegate.kt:52)
	at mozilla.components.feature.contextmenu.DefaultSelectionActionDelegateTest.is call available when passed in and call text selected(DefaultSelectionActionDelegateTest.kt:88)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
```

**Comment by csadilek** [comment]
> It seems like the tests don't like Patterns.EMAIL_ADDRESS.matcher(selectedText).matches()

@sblatz You just need a `@RunWith(AndroidJUnit4::class)` on your test class.

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/Context.kt [hunk 1]
```diff
@@ -100,6 +106,68 @@ fun Context.share(text: String, subject: String = getString(R.string.mozac_suppo
...
+            intent,
+            getString(R.string.mozac_support_ktx_menu_email_with)
+        ).apply {
+            flags = FLAG_ACTIVITY_NEW_TASK
+        }
+
+        startActivity(emailIntent)
+        true
+    } catch (e: ActivityNotFoundException) {
+        Log.log(Log.Priority.WARN, message = "No activity to share to found", throwable = e, tag = "Reference-Browser")
```
Comment (csadilek) [inline components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/Context.kt:135]: nit: `to found`

Comment (csadilek) [reply components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/Context.kt:135]: tag shouldn't be "Reference-Browser"

File: components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/Context.kt [hunk 2]
```diff
@@ -100,6 +106,68 @@ fun Context.share(text: String, subject: String = getString(R.string.mozac_suppo
...
+            intent,
+            getString(R.string.mozac_support_ktx_menu_call_with)
+        ).apply {
+            flags = FLAG_ACTIVITY_NEW_TASK
+        }
+
+        startActivity(callIntent)
+        true
+    } catch (e: ActivityNotFoundException) {
+        Log.log(Log.Priority.WARN, message = "No activity to share to found", throwable = e, tag = "Reference-Browser")
```
Comment (csadilek) [inline components/support/ktx/src/main/java/mozilla/components/support/ktx/android/content/Context.kt:166]: nit: `to found`. tag shouldn't be "Reference-Browser"

File: components/support/ktx/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -6,6 +6,8 @@
   -->
 
 <resources>
+    <string name="mozac_support_ktx_menu_call_with">Call with…</string>
```
Comment (Amejia481) [inline components/support/ktx/src/main/res/values/strings.xml:9]: Let's add translators comments :)

File: samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt [hunk 4]
```diff
@@ -73,11 +75,12 @@ open class BrowserActivity : AppCompatActivity(), ComponentCallbacks2 {
...
-                    components.store,
-                    context
-                ) {
-                    share(it)
-                }
+                    store = components.store,
+                    context = context,
+                    shareTextClicked = { share(it) },
+                    emailTextClicked = { email(it) },
+                    callTextClicked = { call(it) }
```
Comment (Amejia481) [inline samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt:82]: Those can be the default values on `DefaultSelectionActionDelegate` as most of the people will do that :) wdyt?

Comment (sblatz) [reply samples/browser/src/main/java/org/mozilla/samples/browser/BrowserActivity.kt]: Done!

**Comment by sblatz** [comment]
bors r=Amejia481

**Comment by sblatz** [comment]
bors retry

**Pull request merged**